trade_client.py

```diff
@@ -1,34 +1,43 @@
+from datetime import datetime
 from logger import logger
 
 from auth.gateio_auth import *
-from gate_api import ApiClient, Configuration, Order, SpotApi
-from gate_api.exceptions import ApiException, GateApiException
-
-from store_order import store_order
+from gate_api import ApiClient, Order, SpotApi
 
 client = load_gateio_creds('auth/auth.yml')
 spot_api = SpotApi(ApiClient(client))
-import json
 
+current_last_trade = None
 
 
-def get_last_price(base,quote, return_price_only):
+def get_last_price(base, quote, return_price_only):
     """
     Args:
     'DOT', 'USDT'
     """
-    tickers = spot_api.list_tickers(currency_pair=f'{base}_{quote}')
-    assert len(tickers) == 1
-    t = tickers[0]
+    global current_last_trade
+    trades = spot_api.list_trades(currency_pair=f'{base}_{quote}', limit=1)
+    assert len(trades) == 1
+    t = trades[0]
+
+    create_time_ms = datetime.utcfromtimestamp(int(t.create_time_ms.split('.')[0]) / 1000)
+    create_time_formatted = create_time_ms.strftime('%d-%m-%y %H:%M:%S.%f')
+
+    if current_last_trade and current_last_trade.id > t.id:
+        logger.debug(f"STALE TRADEBOOK RESULT FOUND. RE-TRYING.")
+        return get_last_price(base=base, quote=quote, return_price_only=return_price_only)
+    else:
+        current_last_trade = t
+
     if return_price_only:
-        return t.last
-    
-    logger.info(f"GET PRICE: {t.currency_pair} | last={t.last} | change%={t.change_percentage} | lowest_ask={t.lowest_ask} | highest_bid={t.highest_bid} | base_volue={t.base_volume} | quote_volume={t.quote_volume}")
+        return t.price
+
+    logger.info(f"LATEST TRADE: {t.currency_pair} | id={t.id} | create_time={create_time_formatted} | "
```

Comment:
In case `current_last_trade.id > t.id` is true, `t` is actually not the last trade but it's logged as the last trade here?

Reply:
Unsure what you mean. If `current_last_trade.id > t.id` is True, it calls itself recursively and returns the result of that call

Reply:
Ah, perfect! I didn't pay attention to the fact that we're in exactly that function and overlooked the recursion. 👍